Model/AdyenInitPayments.php

```diff
@@ -131,10 +131,10 @@ public function execute(int $adyenCartId, string $stateData): string
                 __('Payments call failed because stateData was not a valid JSON!')
             );
         }
-
+        $amount = $stateData['amount'];
         // Validate the keys in stateData and remove invalid keys
         $stateData = $this->checkoutStateDataValidator->getValidatedAdditionalData($stateData);
-        $paymentsRequest = $this->buildPaymentsRequest($quote, $stateData);
+        $paymentsRequest = $this->buildPaymentsRequest($quote, $stateData, $amount);
```

Comment:
Passing the `amount` from the frontend might create an insecure implementation. Besides that, `ChargedCurrency` helper generates the correct amount depending on the `display` or `base` currency equivalent. Is there a specific reason for not using the quote for `/payments` call and passing if from frontend?